prog/minimization.go

```diff
@@ -121,11 +121,21 @@ func (typ *UnionType) minimize(ctx *minimizeArgsCtx, arg Arg, path string) bool
 }
 
 func (typ *PtrType) minimize(ctx *minimizeArgsCtx, arg Arg, path string) bool {
-	// TODO: try to remove optional ptrs
 	a := arg.(*PointerArg)
 	if a.Res == nil {
 		return false
 	}
+	if !ctx.triedPaths[path] {
+		r0 := a.Res
+		a.Res = nil
+		removeArg(r0)
+		ctx.target.assignSizesCall(ctx.call)
+		if ctx.pred(ctx.p, ctx.callIndex0) {
+			*ctx.p0 = ctx.p
+		}
+		ctx.triedPaths[path] = true
```

Comment:
If we set ctx.triedPaths[path], then we will never try to minimize this arg again, including what it points to. But we do want to still minimize what it points to. Am I missing something? Try to add a test where predicate says to not set pointer to NULL, but we still want to minimize what it points to.

Reply:
See the comment above. Added a test.